promql/parse.go

```diff
@@ -101,7 +101,7 @@ func (p *parser) parseExpr() (expr Expr, err error) {
 			continue
 		}
 		if expr != nil {
-			p.errorf("expression read but input remaining")
+			p.errorf("remaining input %q could not be parsed", p.lex.input[p.lex.lastPos:])
```

Comment:
I'd put the remaining input at the end, in case it's long: "failed to parse additional input after expression: %q"